### PR TITLE
DDPB-4132: Update ClientBenefitsCheck validation errors with new content

### PR DESCRIPTION
**Pull request by acsauk**
## Purpose
Adds content for the various form validation errors we put in place in previous tickets.

Fixes DDPB-4132

## Approach
I'd initially planned to add some of the basic validation by hooking into the in built types provided by Symfony but then as the translations and logic started to get spread around different parts of code I moved it all into the custom validator instead so its easier to see in one place.

## Learning
Testing this was initially a bit of a pain but then I checked out what @aiesoftware had put in place for `CostBreakdownNotGreaterThanTotalValidatorTest` and things fell into place. The initialize step was the missing piece of getting the context object ready for use in a test.

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [x] Translations are used and the profiler doesn't identify any missing
- [ ] Any links or buttons added are screen reader friendly and contextually complete
- [ ] If adding GA events, I have updated or [checked](docs/runbooks/GOOGLE-ANALYTICS.md) the existing category or label values

### CODE REVIEW

File: client/translations/report-client-benefits-check.en.yml [hunk 1]
```diff
@@ -52,6 +52,13 @@ form:
             neverChecked: I've never checked this
         dateCheckedHint: Date checked
         neverCheckedHint: Please tell us why you have never checked this
+        errors:
+            noOptionSelected: Tell us if you have checked %client%'s benefits entitlement
+            missingDate: Enter the date you last checked %client%'s benefits
+            invalidDate: Enter a real date
```
Comment (coderste) [inline client/translations/report-client-benefits-check.en.yml:58]: Maybe it's just me but I feel like this error message is a bit forceful? What does the date input field look like? Is it the DD/MM/YYYY format?

If so then I imagine someone entering `35/13/2015` would return the error `Enter a real date` and while the error message is correct that this is not a real date I think phrasing it like `Date invalid. Please enter a valid date.` would feel a little less forceful maybe

Comment (acsauk) [reply client/translations/report-client-benefits-check.en.yml:58]: Yeah that's a fair point. The content all came from Uchenna so I'd assumed it was following MoJ/GDS style approaches but will raise with Stacey to re-confirm its what we want.

Comment (coderste) [reply client/translations/report-client-benefits-check.en.yml:58]: Ah fair enough yeah if it's part of the MoJ/GDS style then obviously it is what it is just felt a little forceful reading it myself 😆

**Review by coderste — DISMISSED** [review]

File: api/tests/Behat/features-v2/reporting/sections/client-benefits-check/client-benefits-check.lay.feature [hunk 2]
```diff
@@ -114,7 +114,7 @@ Feature: Client benefits check - Lay users
         When I visit the report overview page
         Then I should see "client-benefits-check" as "not finished"
 
-    @lay-combined-high-not-started
+    @lay-combined-high-not-started @acs
```
Comment (coderste) [inline api/tests/Behat/features-v2/reporting/sections/client-benefits-check/client-benefits-check.lay.feature:117]: Does @acs need to be removed?

File: client/assets/javascripts/__tests__/modules/ConditionalFieldRevealer.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { describe, it } from '@jest/globals'
```
Comment (coderste) [inline client/assets/javascripts/__tests__/modules/ConditionalFieldRevealer.test.js:1]: Nice 👏🏻

**Review by coderste — CHANGES_REQUESTED** [review]

**Review by coderste — APPROVED** [review]

**Pull request merged**